nibabel/streamlines/__init__.py

```diff
@@ -0,0 +1,137 @@
+import os
+import warnings
+from ..externals.six import string_types
+
+from .header import Field
+from .array_sequence import ArraySequence
+from .tractogram import Tractogram, LazyTractogram
+from .tractogram_file import ExtensionWarning
+
+from .trk import TrkFile
+
+# List of all supported formats
+FORMATS = {".trk": TrkFile}
+
+
+def is_supported(fileobj):
+    """ Checks if the file-like object if supported by NiBabel.
+
+    Parameters
+    ----------
+    fileobj : string or file-like object
+        If string, a filename; otherwise an open file-like object pointing
+        to a streamlines file (and ready to read from the beginning of the
+        header)
+
+    Returns
+    -------
+    is_supported : boolean
+
+    """
+    return detect_format(fileobj) is not None
+
+
+def detect_format(fileobj):
+    """ Returns the StreamlinesFile object guessed from the file-like object.
+
+    Parameters
+    ----------
+    fileobj : string or file-like object
+        If string, a filename; otherwise an open file-like object pointing
+        to a tractogram file (and ready to read from the beginning of the
+        header)
+
+    Returns
+    -------
+    tractogram_file : :class:`TractogramFile` class
+        The class type guessed from the content of `fileobj`.
+
+    """
+    for format in FORMATS.values():
+        try:
+            if format.is_correct_format(fileobj):
+                return format
+
+        except IOError:
+            pass
+
+    if isinstance(fileobj, string_types):
+        _, ext = os.path.splitext(fileobj)
+        return FORMATS.get(ext.lower())
+
+    return None
+
+
+def load(fileobj, lazy_load=False, ref=None):
+    """ Loads streamlines in *RAS+* and *mm* space from a file-like object.
+
+    Parameters
+    ----------
+    fileobj : string or file-like object
+        If string, a filename; otherwise an open file-like object
+        pointing to a streamlines file (and ready to read from the beginning
+        of the streamlines file's header).
+    lazy_load : {False, True}, optional
+        If True, load streamlines in a lazy manner i.e. they will not be kept
+        in memory. Otherwise, load all streamlines in memory.
+
+    Returns
+    -------
+    tractogram_file : :class:TractogramFile object
+        Returns an instance of a :class:TractogramFile containing data and
+        metadata of the tractogram loaded from `fileobj`.
+
+    Notes
+    -----
+    The streamline coordinate (0,0,0) refers to the center of the voxel.
+
```

Comment:
No problem. I wasn't sure which convention you were using. I'm working on different libraries, each one having their own, :-/. I think Numpy (the one adopted by Dipy... I think) requires a blank line but maybe not PEP8.

On a side note, do you enforce 72 characters for __docstring__?